test.js

```diff
@@ -690,6 +690,27 @@ test('uses the custom successMessage callback if passed in as an option', functi
   })
 })
 
+test('custom successMessage has optional error parameter', function (t) {
+  var dest = split(JSON.parse)
+  var customErrorMessage = 'Custom response message'
+  var logger = pinoHttp({
+    customSuccessMessage: function (res, err) {
+      var optionalErrorMessage = err ? err.toString() : 'error is undefined'
+      return customErrorMessage + ' ' + optionalErrorMessage
+    }
+  }, dest)
+
+  setup(t, logger, function (err, server) {
+    t.error(err)
+    doGet(server, '/404')
+  })
+
+  dest.on('data', function (line) {
+    t.contains(line.msg, customErrorMessage)
```

Comment:
```suggestion
    t.is(line.msg, `${customErrorMessage} <the text from the error message>`)
```

Reply:
Yes you're right, this test would not make much sense without testing for an actual error. I used the logic from the original implementation of `customErrorMessage` and it seems like this test is not making any sense either as its only checking for the initial message too:
https://github.com/pinojs/pino-http/blob/4d1c32038c8d287befe41f8be096047b77b1435f/test.js#L693-L711

I tried adding a 4xx error but I'm not quite sure how to achieve it as something like this:
```JS
  function handle (req, res) {
    logger(req, res)
    res.emit('error', new Error('Some error'))
    res.end()
  }

  setup(t, logger, function (err, server) {
    t.error(err)
    doGet(server, '/404')
  }, handle)
```
would just run into the default error handling and respond with a `request errored`.

The test is making me question if this PR is really making any sense as I'm not quite sure if the 4xx errors I wanted to handle initially with this implementation are more of an issue to solve in the application framework where I am consuming pino-http.

Reply:
So what is your conclusion?